src/MyConnection.h

```diff
@@ -18,10 +18,15 @@ class MyConnection : boost::noncopyable {
 
 public:
 
-  MyConnection(std::string host, std::string user, std::string password,
-               std::string db, unsigned int port, std::string unix_socket,
-               unsigned long client_flag, std::string groups,
-               std::string default_file) :
+  MyConnection(Rcpp::Nullable < std::string > host,
```

Comment:
Thanks for the investigation. Are you OK with leaving it "as is" until Rcpp is fixed? It would be great if you could double-check that my PR there actually solves the issue at hand.